crates/osutils/src/netplan.rs

```diff
@@ -42,6 +53,37 @@ fn render_netplan_yaml(value: &NetworkConfig) -> Result<String, Error> {
         .context("Failed to render netplan yaml")
 }
 
+/// Backs up the current netplan config.
+pub fn backup() -> Result<(), Error> {
+    debug!("Backing up current state of netplan config");
+    // Ensure backup directory is empty beforehand.
+    cleanup_backup()?;
+    // Re-create the directory.
+    if let Some(parent) = Path::new(NETPLAN_BACKUP_FILE).parent() {
+        fs::create_dir_all(parent).with_context(|| {
+            format!(
+                "Failed to create netplan config backup directory at '{}'",
+                parent.display()
+            )
+        })?;
+    }
+    // Copy over existing netplan config if it exists.
```

Comment:
i'm curious what happens if there is another netplan file in the netplan folder (if a customer creates one with additionalFiles or something).